source/core/TDCodecVia.cpp

```diff
@@ -500,23 +500,30 @@ void TDViaParser::ParseAggregateElementList(std::vector<TypeRef> *elementTypesVe
     SubString  token;
     SubString  fieldName;
     UsageTypeEnum  usageType;
+    bool isDataItem;
 
     _string.ReadToken(&token);
     if (!token.CompareCStr("("))
         return LOG_EVENT(kHardDataError, "'(' missing");
 
     _string.ReadToken(&token);
     while (!token.CompareCStr(")")) {
+        isDataItem = false;
         if (token.CompareCStr(tsElementToken)) {
             usageType = kUsageTypeSimple;
+        } else if (token.CompareCStr(tsDataitemElementToken)) {
+            usageType = kUsageTypeSimple;
+            isDataItem = true;
         } else if (token.CompareCStr(tsConstElementToken)) {
             usageType = kUsageTypeConst;
         } else if (token.CompareCStr(tsInputParamToken)) {
             usageType = kUsageTypeInput;
         } else if (token.CompareCStr(tsOutputParamToken)) {
             usageType = kUsageTypeOutput;
+            isDataItem = true;
```

Comment:
Let's check for "d" in Params block also to mark isDataItem true for completeness (as we do for data Items in local variables). It is a good idea to make DFIR generate VIA to indicate which variables are data items regardless of which block it belongs to (Locals or Params). In the future, there are potentially other attributes/meta-data DFIR may want to put in VIA for variables or instructions that Vireo can make use of. It is better to parse that directly in Vireo. I made the same comment in @ggumen 's DFIR side PR too. We all can discuss in person if we want to.

Reply:
That doesn't make sense, because Param blocks can only contain inputs and outputs.  Otherwise they are not 'parameters'.

Reply:
Also, the reason inputs are not auto-marked dataItems is because you cannot write to an input in the first place; a Copy to it would generate an error.
My hunch is that if you duplicate an input terminal and write to it, it because 'io' in DFIR for Vireo. in which case it would work.  Ask Gandi about this.

Reply:
Currently, we do not emit 'io' in VIA from DFIR yet. Sankara tried to put that change in last release, but had to revert it before we shipped. 

To me the naming of this variable `isDataItem` is confusing. DataItems in LabVIEW can be either Control or Indicator, and whether it's a Control or an Indicator, it can always be read and written to through its multiple DataAccessors. 
The concept of vireo `io` doesn't match LabVIEW's multiple DataAccessors. We should change Vireo to accept copying to `i` DataItems from its DataAccessor local. I think that's why Sankara had to revert his change.

Maybe we want to call this variable `isIndicatorDataItem`?

Reply:
So I infer from this that having a write DataAccessor for a dataItem that is an input on the conpane will currently NOT work in Vireo (already, regardless of any changes made in this PR)?

Reply:
It feels fundamentally wrong to me for Vireo to allow writing to inputs.  If it's written to, it should be 'io', and I don't understand why this doesn't match multiple data accessors.  The use-case seems the same to me.

Reply:
We discussed this in person. "io" should be used only in the cases where a callee's modification should reflect in the caller. We need to support in NXG (for Web target) allowing writing to duplicate terminal of control in the connector pane in the future and probably not urgent. I think the notion of "i", "o", "io" with respect to the contract between callee/caller should be distinguished from the notion of mutability of the same in the SubVI. 

DFIR marking only items in local as data item and not the items on the param block and then Vireo not marking only input data items in Param block as not a data item is inconsistent and not clear.

Reply:
I just updated so inputs are also considered dataItems by Vireo, even though they don't need to check for updateNeeded, for consistency.